src/pgwire/src/server.rs

```diff
@@ -139,7 +144,14 @@ impl Server {
                                     internal,
                                     active_connection_count,
                                 })
-                                .await?;
+                                .await;
+
+                                match &conn_res {
+                                    Ok(_) => debug!(conn_uuid = %conn_uuid.display_or("<none>"), "closing pgwire connection in adapter successfully"),
+                                    Err(e) => debug!(conn_uuid = %conn_uuid.display_or("<none>"), "closing pgwire connection in adapter with error {e}"),
+                                }
+
+                                conn_res?;
                                 conn.flush().await?;
```

Comment:
Maybe a little sketchy that we return an error before attempting to flush? I'll look into this later.

Reply:
IIRC, and error here indicates something went wrong with I/O, and so flushing isn't likely to work—the socket is already broken.